### PR TITLE
Add VS Code How To Guide

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Partly Adresses #2833

By accident builds on top of #5112 (`load.md` file). I don't know how to remove that.

Preview: [vscode_configure.md](https://github.com/holoviz/panel/blob/enhancement/vs-code-docs/doc/how_to/editor/vscode_configure.md),  [editor_debug.md](https://github.com/holoviz/panel/blob/enhancement/vs-code-docs/doc/how_to/editor/editor_debug.md), [editor.md](https://github.com/holoviz/panel/blob/enhancement/vs-code-docs/doc/how_to/editor/editor.md), 

## Todo

- [x] Write proposal
- [x] Split into more dedicated guies
- [ ] Get review
- [ ] Finalize

## Maybe later

- [ ] how to use Panel with VS Code dark themes in notebook environment. 
- [ ] how to debug bound Panel callbacks in notebook environment
- [ ] Figure out how to add keyboard shortcut to open `https://panel.holoviz.org`. I could not figure it out.
- [ ] Create really big and useful *snippets* file instead of just a few examples.
- [ ] Add similar guide(s) for PyCharm.

## Issues identified

- [ ] Panel not working with dark theme in notebook environment (or at least I could not figure out get something readable)

### CODE REVIEW

File: doc/_static/vscode-notebook.png
Comment (MarcSkovMadsen) [inline doc/_static/vscode-notebook.png:1]: I could not get this working with dark theme. The text in the output cells is not readable.

File: doc/how_to/editor/vscode.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            "name": "panel serve",
+            "type": "python",
+            "request": "launch",
+            "program": "-m",
+            "args": [
+                "panel",
+                "serve",
+                "${relativeFile}",
+                "--show"
```
Comment (MarcSkovMadsen) [inline doc/how_to/editor/vscode.md:48]: Maybe we should add `--autoreload` also. But I have no experience telling me if this is a good idea or not.

**Comment by MarcSkovMadsen** [comment]
I've split refactored such that the material is split into

- The existing `editor.md` guide
- A new `editor_debug.md` guide
- A new `vscode_configure` guide.

To me that makes sense. Let me know what you think @philippjfr . Thanks.

**Comment by MarcSkovMadsen** [comment]
This one is ready for review. Don't plan to do more before the review.

**Review by philippjfr — APPROVED** [review]
Thanks, this is already looking good. We can always tweak more later but we might as well get this into 1.2.1.

**Pull request merged**